src/main/webapp/js/science_portal.js

```diff
@@ -307,14 +327,8 @@
     }
 
     function setFormFields(sessionType) {
-      // go through _sessionTypeMap.session_types
-      var formList = []
-      for (var i = 0; i < _sessionTypeMap.session_types.length; i++) {
-        if (_sessionTypeMap.session_types[i].name === sessionType) {
-          formList = _sessionTypeMap.session_types[i].form_fields
-          break
-        }
-       }
+      var mapEntry = getMapEntry(sessionType)
+      var formList = mapEntry.form_fields
```

Comment:
`mapEntry` can be `null` here.

Reply:
Added a catch for this, and a message - if the config is messed up that badly at this point in the code, the page should be refreshed.   Functionally the mapEntry won't be null because the values from the session map are used to populate the dropdown that has the listener function that eventually leads to this bit of code... But yes, there should be a check for null here.